src/Common/MemoryTracker.h

```diff
@@ -142,6 +142,19 @@ class MemoryTracker
         ~BlockerInThread() { is_blocked = false; }
         static bool isBlocked() { return is_blocked; }
     };
+
+    /// To be able to avoid MEMORY_LIMIT_EXCEEDED Exception in destructors.
+    struct LockExceptionInThread
```

Comment:
This does not used in current PR, this was added only to add ability to explicitly disable exceptions in MemoryTracker.

@alexey-milovidov what do you think, should it be kept or removed and added only once it will be required?

Reply:
We can keep for now and maybe remove later if unused.

Reply:
Ok, I've added some comments and rebase